### PR TITLE
Add a win condition and print a victory message

### PR DESCRIPTION
**Pull request by mystal**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -193,13 +191,24 @@ impl<'a, 'b> ggez::event::EventHandler for State<'a, 'b> {
...
-                ctx,
-                &graphics::Text::new("Hey. I'm sorry to tell you this. You died."),
-                graphics::DrawParam::default()
-                    .dest([10.0, 10.0]),
-            )?;
+        match *play_state {
+            PlayState::Win => {
+                graphics::draw(
+                    ctx,
+                    &graphics::Text::new("VICTORY ACHIEVED!"),
```
Comment (Ablexxive) [inline src/main.rs:198]: I hear this in the voice of the Protoss from SC1

File: src/resources.rs [hunk 2]
```diff
@@ -1,10 +1,20 @@
 use specs::Entity;
 
-#[derive(Default)]
+#[derive(Clone, Copy, Debug, Default)]
 pub struct DeltaTime(pub f32);
 
-#[derive(Default)]
-pub struct YouLose(pub bool);
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PlayState {
```
Comment (Ablexxive) [inline src/resources.rs:7]: Nice!

File: src/systems.rs [hunk 3]
```diff
@@ -262,16 +262,36 @@ impl<'a> System<'a> for DeathSystem {
...
+        ReadStorage<'a, Enemy>,
+        ReadStorage<'a, Spawner>,
+        Write<'a, PlayState>,
+    );
+
+    fn run (&mut self, data: Self::SystemData) {
+        let (bases, enemies, spawners, mut play_state) = data;
+
+        // Player wins When all enemies and spawners are gone and there are still bases alive.
+        if !bases.is_empty() && enemies.is_empty() && spawners.is_empty() {
```
Comment (Ablexxive) [inline src/systems.rs:293]: Oh I didn't know about the `is_empty()` fcn, thats great

**Review by Ablexxive — APPROVED** [review]

File: src/systems.rs (hunk 3 above)
Comment (mystal) [reply src/systems.rs:293]: Yeah, I was like, "surely this has to exist" and dug around the specs documentation. https://docs.rs/specs/0.15.0/specs/storage/struct.Storage.html#method.is_empty

**Pull request merged**